community/kernel/src/test/java/org/neo4j/kernel/impl/store/record/SchemaRuleSerializationTest.java

```diff
@@ -36,48 +37,108 @@
 {
     private static final int VERY_LARGE_CAPACITY = 1024;
 
+    IndexRule indexRegular = IndexRule.indexRule( RULE_ID,
+            NewIndexDescriptorFactory.forLabel( LABEL_ID, PROPERTY_ID_1 ), PROVIDER_DESCRIPTOR );
+
+    IndexRule indexUnique = IndexRule.constraintIndexRule( RULE_ID_2,
+            NewIndexDescriptorFactory.uniqueForLabel( LABEL_ID, PROPERTY_ID_1 ), PROVIDER_DESCRIPTOR, RULE_ID );
+
+    IndexRule indexCompositeRegular = IndexRule.indexRule( RULE_ID,
+            NewIndexDescriptorFactory.forLabel( LABEL_ID, PROPERTY_ID_1, PROPERTY_ID_2 ), PROVIDER_DESCRIPTOR );
+
+    IndexRule indexCompositeUnique = IndexRule.constraintIndexRule( RULE_ID_2,
+            NewIndexDescriptorFactory.uniqueForLabel( LABEL_ID, PROPERTY_ID_1, PROPERTY_ID_2 ),
+            PROVIDER_DESCRIPTOR, RULE_ID );
+
+    IndexRule indexBigComposite = IndexRule.indexRule( RULE_ID,
+            NewIndexDescriptorFactory.forLabel( LABEL_ID, IntStream.range(1, 200).toArray() ), PROVIDER_DESCRIPTOR );
+
+    ConstraintRule constraintExistsLabel = ConstraintRule.constraintRule( RULE_ID,
+            ConstraintDescriptorFactory.existsForLabel( LABEL_ID, PROPERTY_ID_1 ) );
+
+    ConstraintRule constraintUniqueLabel = ConstraintRule.constraintRule( RULE_ID_2,
+            ConstraintDescriptorFactory.uniqueForLabel( LABEL_ID, PROPERTY_ID_1 ), RULE_ID );
+
+    ConstraintRule constraintExistsRelType = ConstraintRule.constraintRule( RULE_ID_2,
+            ConstraintDescriptorFactory.existsForRelType( REL_TYPE_ID, PROPERTY_ID_1 ) );
+
+    ConstraintRule constraintUniqueRelType = ConstraintRule.constraintRule( RULE_ID,
+            ConstraintDescriptorFactory.uniqueForRelType( REL_TYPE_ID, PROPERTY_ID_1 ), RULE_ID_2 );
+
+    ConstraintRule constraintCompositeLabel = ConstraintRule.constraintRule( RULE_ID,
+            ConstraintDescriptorFactory.existsForLabel( LABEL_ID, PROPERTY_ID_1, PROPERTY_ID_2 ) );
+
+    ConstraintRule constraintCompositeRelType = ConstraintRule.constraintRule( RULE_ID_2,
+            ConstraintDescriptorFactory.existsForRelType( REL_TYPE_ID, PROPERTY_ID_1, PROPERTY_ID_2 ) );
```

Comment:
But again, it'd be nice to have buffers of pre-serialised records to regression-test with.